src/AdsConsent.ts

```diff
@@ -75,53 +71,53 @@ export const AdsConsent: AdsConsentInterface = {
     return native.requestInfoUpdate(options);
   },
 
-  showForm(): Promise<AdsConsentInfo> {
+  showForm() {
```

Comment:
interesting, so these are no longer async?
That's a breaking change for people doing old .then/.catch/etc style, but won't matter for people doing async/await (other than lint stuff)

Reply:
I might just be misreading the type, in that now maybe typescript is inferring things and just infers the type correctly, such that it is async and is typed as described later on where I made my second comment...

Reply:
The type can now be inferred indeed and that's why I removed them.

Reply:
Types are now defined in the spec (NativeConsentModule.ts) and automatically inferred by the native layer. This is thanks to the use of the React Native Codegen tool, which bridges the gap between TypeScript types and the native implementation.

Reply:
awesome, resolved on both then - thanks!